subsys/net/lib/lwm2m/lwm2m_shell.c

```diff
@@ -71,9 +89,192 @@ static int cmd_send(const struct shell *sh, size_t argc, char **argv)
 	return 0;
 }
 
+static int cmd_exec(const struct shell *sh, size_t argc, char **argv) 
+{
+	struct lwm2m_ctx *ctx = lwm2m_rd_client_ctx();
+
+	if (!ctx) {
+		shell_error(sh, "no lwm2m context yet\n");
+		return -ENOEXEC;
+	}
+
+	int ignore_cnt = 2;	/* Subcmd + PATH */
+	const char* pathstr = argv[1];
+	struct lwm2m_engine_res *res = NULL;
+	int ret = lwm2m_engine_get_resource(pathstr, &res);
+	if (ret < 0)
+	{
+		shell_error(sh, "Couldn't execute resource\n");
+		return ret;
+	}
+	if(res->execute_cb)
+	{
+		uint16_t obj_id;
+		sscanf(pathstr, "%*d/%hd/%*d", &obj_id);
+		return res->execute_cb(obj_id, argv[ignore_cnt], argc-ignore_cnt);
+	}
+	return 0;
+}
+
+static int cmd_read(const struct shell *sh, size_t argc, char **argv) 
+{
+	
+	struct lwm2m_ctx *ctx = lwm2m_rd_client_ctx();
+	if (!ctx) {
+		shell_error(sh, "no lwm2m context yet\n");
+		return -ENOEXEC;
+	}
+
+	if (argc < 2) {
+		shell_error(sh, "no arguments or path(s)\n");
+		shell_help(sh);
+		return -EINVAL;
+	}
+
+	const char* dtype = "-s"; /* default */
+	if (argc > 2) { /* read + path + options(data type) */
+		dtype = argv[2];
+	}
+
+	const char* pathstr = argv[1];
+	int ret = 0;
+	
+	if (strcmp(dtype, "-s") == 0){
+		const char* buff;
+		uint16_t buff_len=0;
+		uint8_t buff_flags;
+		ret = lwm2m_engine_get_res_data(pathstr, (void **)&buff, &buff_len, &buff_flags);
```

Comment:
Zephyr coding style requires you to have one empty line between variable introduction and the next line of code.

```suggestion
		const char* buff;
		uint16_t buff_len=0;
		uint8_t buff_flags;

		ret = lwm2m_engine_get_res_data(pathstr, (void **)&buff, &buff_len, &buff_flags);
```

This this similarly to rest of the cases as well.